### PR TITLE
Introduce --main flag

### LINKED ISSUE #1018: CLI spends a long time scanning JAR file to find main for Java actions
If one creates a Java action from a JAR containing a large number of classes, the time it takes the CLI to "create" the action starts to be come unacceptably long. The CLI runs `javap` for each class to find `main` and each of these can take a second or so. So it adds up fast. The CLI could give up once it find 1 match (rather than look for all), or allow/require the user to specify their `package.class.method` for `"main"`.

Related to #47.
FYI @psuter @mdeuser @dubeejw.

### LINKED ISSUE #1026: Unable to use the same jar in multiple Java actions
I would like to use the same jar for two Java actions, each with a slightly different 'main' class/method (but sharing the rest of the jar).
Today (with wsk CLI) this does not seem possible, because I can pass only one JAR (which must include all the dependencies etc), and the CLI 'guesses' the main class (by analyzing the jar and picking the first class with proper 'main' method signature).

Related to #1018

### PR DESCRIPTION
**Pull request by psuter**
Introduces a `--main` flag in the CLI for action creation and update.

For Java, the flag is required; the CLI no longer attempts to determine the action entry point by analyzing the JAR file, which was potentially slow and error prone.

Users can now also create multiple Java actions using the same JAR file but a different entry point.

Fixes #1018.
Fixes #1026.

For NodeJS, Swift, Python, the flag lets the user optionally chose an alternative entry point than `main`.

Fixes (most of) #47.

### CODE REVIEW

**Comment by rabbah** [comment]
:+1: 

We should ensure that the invoker will always include the "main" metadata on `init` and shift the "main" defined check there (although defensively coding against it in the action runner/proxy is ok).

**Comment by psuter** [comment]
I've reached the set of features I wanted. Opening up for (rest of) review.

**Comment by psuter** [comment]
Rebased. FYI `d96fe1d` has become 116684478e8f03a552de762b3519b15b6fc7b5a4.

**Comment by psuter** [comment]
PG3 37 is blue.

File: tests/src/system/basic/CLIJavaTests.scala [hunk 1]
```diff
@@ -46,11 +47,13 @@ class CLIJavaTests
     /**
      * Test the Java "hello world" demo sequence
      */
-    it should "invoke a java action" in withAssetCleaner(wskprops) {
+    it should "Invoke a java action" in withAssetCleaner(wskprops) {
```
Comment (rabbah) [inline tests/src/system/basic/CLIJavaTests.scala]: ah man - we'll have to rename all our tests now :)

Comment (psuter) [inline tests/src/system/basic/CLIJavaTests.scala]: I like to focus on the essential.

**Comment by psuter** [comment]
Rebased again, and triggered builds for the "critical" commits (the ones with rebase conflicts). When all is green, I suggest we squash the first two commits into one, and the last 3 into one. This will nicely separate the work by language and components (containers first, CLI+entities then).

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/compare/da7ac1328e92...d98188d7eb74.

**Pull request closed without merging**